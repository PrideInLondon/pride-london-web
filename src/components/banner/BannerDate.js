import React from 'react'
import PropTypes from 'prop-types'
import { StyledBannerDate, BannerDateWrapper } from './BannerDate.styles'

const BannerDate = ({ children }) => (
  <BannerDateWrapper>
    <StyledBannerDate>{children}</StyledBannerDate>
  </BannerDateWrapper>
)

BannerDate.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
}

export default BannerDate
