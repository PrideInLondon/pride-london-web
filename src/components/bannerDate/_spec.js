import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow } from 'enzyme'
import BannerDate from './'
import 'jest-styled-components'

describe('<BannerDate/>', () => {
  it('renders', () => {
    const wrapper = shallow(<BannerDate>foo</BannerDate>)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('renders a <p/>', () => {
    const text = 'foo'
    const wrapper = shallow(<BannerDate>{text}</BannerDate>)
    expect(wrapper.find('bannerDate__BannerDate')).toHaveLength(1)
    expect(wrapper.find('bannerDate__BannerDate').text()).toBe(text)
  })
})
