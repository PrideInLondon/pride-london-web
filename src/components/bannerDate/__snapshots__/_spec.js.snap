// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerDate/> renders 1`] = `
<bannerDate__BannerDateWrapper>
  <bannerDate__BannerDate>
    foo
  </bannerDate__BannerDate>
</bannerDate__BannerDateWrapper>
`;
